src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,280 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import {Ownable} from "@openzeppelin/contracts-0.8/access/Ownable.sol";
+import {Address} from "@openzeppelin/contracts-0.8/utils/Address.sol";
+import {Math} from "@openzeppelin/contracts-0.8/utils/math/Math.sol";
+import {IERC721} from "@openzeppelin/contracts-0.8/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts-0.8/token/ERC1155/IERC1155.sol";
+
+contract RequirementsRules is Ownable {
+    using Address for address;
+
+    // masStake amount allowed if user has no ERC721 or ERC1155
+    uint256 public maxStakeOverall;
+
+    struct RequirementRuleERC721 {
```

Comment:
Generally speaking, the order should be:
ERC721RequirementRule

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc